tiledb/sm/c_api/tiledb_filestore.cc

```diff
@@ -270,70 +318,93 @@ TILEDB_EXPORT int32_t tiledb_filestore_uri_import(
        static_cast<uint64_t>(file_size % tile_extent != 0)) *
           tile_extent -
       1;
-  subarray.add_range(0, static_cast<uint64_t>(0), last_space_tile_boundary);
+  uint64_t range_arr[] = {static_cast<uint64_t>(0), last_space_tile_boundary};
+  tiledb::type::Range subarray_range(
+      static_cast<void*>(range_arr), sizeof(uint64_t) * 2);
+  throw_if_not_ok(subarray.add_range(0, std::move(subarray_range)));
   query.set_subarray(subarray);
 
   auto tiledb_cloud_fix = [&](uint64_t start, uint64_t end, uint64_t nbytes) {
```

Comment:
safe to remove `nbytes`, `end - start + 1` should be enough within the function.